frontend/tests/controllers/ContestListv2Test.php

```diff
@@ -256,4 +256,178 @@ public function testPrivateContestsForSystemAdmin() {
+
+        $secondContestCreator = $secondContestData['director'];
+
+        // Logging user
+        $login = self::login($secondContestCreator);
+
+        $contestListPayload = \OmegaUp\Controllers\Contest::getContestListDetailsv2ForTypeScript(
+            new \OmegaUp\Request([
+                'auth_token' => $login->auth_token,
+            ])
+        )['smartyProperties']['payload'];
+
+        $contests = $contestListPayload['contests']['current'];
+        $contestWithoutContestants = $contests[0];
+
+        $this->assertEquals(0, $contestWithoutContestants['contestants']);
+    }
+
+    public function testContestantsColumnAsInvitedUser() {
+        [
+            'firstContestData' => $firstContestData,
+            'secondContestData' => $secondContestData,
+            'firstInvitedUserIdentity' => $firstInvitedUserIdentity,
+            'secondInvitedUserIdentity' => $secondInvitedUserIdentity,
+        ] = $this->createContestsAndAddContestants('private');
+
+        // Logging user
+        $login = self::login($firstInvitedUserIdentity);
+
+        $contestListPayload = \OmegaUp\Controllers\Contest::getContestListDetailsv2ForTypeScript(
+            new \OmegaUp\Request([
+                'auth_token' => $login->auth_token,
+            ])
+        )['smartyProperties']['payload'];
+
+        $contests = $contestListPayload['contests']['current'];
+        $contestWithContestants = $contests[0];
+
+        $this->assertEquals(2, $contestWithContestants['contestants']);
+    }
+
+    public function testContestantsColumnAsSystemAdmin() {
+        [
+            'firstContestData' => $firstContestData,
+            'secondContestData' => $secondContestData,
+            'firstInvitedUserIdentity' => $firstInvitedUserIdentity,
+            'secondInvitedUserIdentity' => $secondInvitedUserIdentity,
+        ] = $this->createContestsAndAddContestants('private');
+
+        // Create admin user (system admin)
+        ['user' => $user, 'identity' => $adminUserIdentity] = \OmegaUp\Test\Factories\User::createAdminUser();
+
+        // Logging user
+        $login = self::login($adminUserIdentity);
+
+        $contestListPayload = \OmegaUp\Controllers\Contest::getContestListDetailsv2ForTypeScript(
+            new \OmegaUp\Request([
+                'auth_token' => $login->auth_token,
+            ])
+        )['smartyProperties']['payload'];
+
+        $contests = $contestListPayload['contests']['current'];
+
+        if ($contests[0]['title'] === 'contest-with-contestants') {
+            $contestWithContestants = $contests[0];
+            $contestWithoutContestants = $contests[1];
+        } else {
+            $contestWithContestants = $contests[1];
+            $contestWithoutContestants = $contests[0];
+        }
+
+        $this->assertEquals(2, $contestWithContestants['contestants']);
+        $this->assertEquals(0, $contestWithoutContestants['contestants']);
```

Comment:
```suggestion
        $contestContestants = [];
        foreach ($contests as $contest) {
            $contestContestants[$contest['title']] = $contest['contestants'];
        }

        $this->assertEqualsCanonicalizing(
            [
                'contest-with-contestants' => 2,
                'contest-without-contestants' => 0,
            ],
            $contestContestants,
        );
```

así es más fácil saber qué se espera Y además de paso se asevera que la lista tenga la longitud correcta. igual en todos los demás casos.

en general, si se tiene que usar un comportamiento distinto para algo en una prueba dependiendo de una cadena (`$contests[0]['title'] === 'contest-with-contestants'`), hay que buscar una alternativa.

Reply:
Va, me late.